crates/bevy_ui/src/layout/mod.rs

```diff
@@ -164,11 +164,25 @@ without UI components as a child of an entity with UI components, results may be
         parent_window: Entity,
         children: impl Iterator<Item = Entity>,
     ) {
-        let taffy_node = self.window_nodes.get(&parent_window).unwrap();
-        let child_nodes = children
+        let parent_node = *self.window_nodes.get(&parent_window).unwrap();
+
+        let new_window_children = children
             .map(|e| *self.entity_to_taffy.get(&e).unwrap())
-            .collect::<Vec<taffy::node::Node>>();
-        self.taffy.set_children(*taffy_node, &child_nodes).unwrap();
+            .collect::<Vec<_>>();
+
+        let window_children = if let Ok(mut window_children) = self.taffy.children(parent_node) {
+            let child_set = new_window_children.iter().copied().collect::<HashSet<_>>();
+            let mut unique = HashSet::new();
+            window_children.extend(new_window_children);
+            window_children.retain(|child| child_set.contains(child) && unique.insert(*child));
+            window_children
+        } else {
+            new_window_children
```

Comment:
Does this branch ever execute in practice? It seems like `taffy.children` unconditionally returns `Ok`.

Reply:
> It seems like taffy.children unconditionally returns Ok

It currently does but arguably shouldn't, as it will panic if the node doesn't exist (note that if the node exists in Taffy, then the children definitely will as children is default initialized to an empty Vec). Likely Taffy's API will split/duplicated into panicking and result-returning variants at some point. See https://github.com/DioxusLabs/taffy/issues/519